db-client-java/src/main/java/com/eventstore/dbclient/AbstractCreatePersistentSubscription.java

```diff
@@ -36,16 +36,16 @@ public CompletableFuture execute() {
 
             settingsBuilder
                     .setResolveLinks(settings.shouldResolveLinkTos())
-                    .setReadBatchSize(settings.getReadBatchSize())
-                    .setMinCheckpointCount(settings.getCheckPointLowerBound())
-                    .setMaxCheckpointCount(settings.getCheckPointUpperBound())
-                    .setMessageTimeoutMs(settings.getMessageTimeoutMs())
-                    .setMaxSubscriberCount(settings.getMaxSubscriberCount())
-                    .setMaxRetryCount(settings.getMaxRetryCount())
-                    .setLiveBufferSize(settings.getLiveBufferSize())
-                    .setHistoryBufferSize(settings.getHistoryBufferSize())
+                    .setReadBatchSize((int)settings.getReadBatchSize())
+                    .setMinCheckpointCount((int)settings.getCheckPointLowerBound())
+                    .setMaxCheckpointCount((int)settings.getCheckPointUpperBound())
+                    .setMessageTimeoutMs((int)settings.getMessageTimeoutMs())
```

Comment:
**CONCERN:** @hayley-jean, do we plan to align that on the proto level? It seems weird to have `long` in RFC, but the server accepting only `int`.